lib/salus/scanners/yarn_audit.rb

```diff
@@ -7,14 +7,84 @@
 
 module Salus::Scanners
   class YarnAudit < NodeAudit
-    AUDIT_COMMAND = 'yarn audit --json'.freeze
+    # the command was previously 'yarn audit --json', which had memory allocation issues
+    # see https://github.com/yarnpkg/yarn/issues/7404
+    AUDIT_COMMAND = 'yarn audit --no-color'.freeze
+    TT = "┌───────────────┬──────────────────────────────────────────────────────────────┐".freeze
+    TB = "└───────────────┴──────────────────────────────────────────────────────────────┘".freeze
 
     def should_run?
       @repository.yarn_lock_present?
     end
 
+    def run
+      shell_return = Dir.chdir(@repository.path_to_repo) do
+        command = "#{AUDIT_COMMAND} #{scan_deps}"
+        shell_return = run_shell(command)
+
+        excpts = @config.fetch('exceptions', []).map { |e| e["advisory_id"].to_i }
+        report_info(:ignored_cves, excpts)
+        return report_success if shell_return.success?
+
+        stdout_lines = shell_return.stdout.split("\n")
+        table_start_pos = stdout_lines.index(TT)
+        table_end_pos = stdout_lines.rindex(TB)
+        table_lines = stdout_lines[table_start_pos..table_end_pos]
```

Comment:
I think you should move this after we check if `table_start_pos.nil?`. Is there a test for the case where there is a `stderr` return but no vulns/tables in the output?

Reply:
FIx the `table_start_pos.nil?`

Reply:
Yes, there is already a test case for the case where there is a stderr return but no vulns/tables in the output.
It is `should fail with error if there are errors` in yarn_audit_spec.rb.